core/src/main/scala/akka/persistence/r2dbc/R2dbcSettings.scala

```diff
@@ -328,6 +324,39 @@ final class PublishEventsDynamicSettings(config: Config) {
   val throughputCollectInterval: FiniteDuration = config.getDuration("throughput-collect-interval").asScala
 }
 
+/**
+ * INTERNAL API
+ */
+@InternalStableApi
+final class CodecSettings(
+    val journalPayloadCodec: PayloadCodec,
+    val snapshotPayloadCodec: PayloadCodec,
+    val durableStatePayloadCodec: PayloadCodec,
+    val tagsCodec: TagsCodec,
+    val timestampCodec: TimestampCodec,
+    val queryAdapter: QueryAdapter) {
+
+  // implicits that can be imported
+  object JournalImplicits {
```

Comment:
Have not seen this very often, objects within a case class. But I guess it is fine. Alternative could be something based on inheritance or so, this this `CodecSettings.this.journalPayloadCodec` could be avoided.

Reply:
CodecSettings.this.journalPayloadCodec was just because I used the same names

Reply:
yeah, the object is only for name scoping